File: Programação Funcional/Básicos/Haskell/07.hs
{- 
	Faça uma função que receba um ângulo a e retorne uma tupla contendo o seno da metade desse ângulo utilizando a identidade +- sqrt((1 - cos x)/2):
-}
import Prelude
import Data.Tuple

-- Função que calcula o seno da metade do ângulo usando a identidade trigonométrica
senoMetadeAngulo :: Floating a => a -> (a, a)
senoMetadeAngulo a = (a, sqrt ((1 - cos a) / 2))


-- Função principal para testar
main :: IO ()
main = do
    print (senoMetadeAngulo 60) -- (60.0,0.9880316240928618)
    print (senoMetadeAngulo 90) -- (90.0,0.8509035245341184)

File: Programação Funcional/Básicos/Scala/07.scala
import scala.math._

object Main {
  // Função que calcula o seno da metade do ângulo usando a identidade trigonométrica
  def senoMetadeAngulo(a: Double): (Double, Double) = {
    (a, sqrt((1 - cos(a)) / 2))
  }

  // Função principal para testar
  def main(args: Array[String]): Unit = {
    println(senoMetadeAngulo(60)) // (60.0,0.8660254037844387)
    println(senoMetadeAngulo(90)) // (90.0,0.7071067811865476)
  }
}
